src/garage/torch/algos/pearl.py

```diff
@@ -1,6 +1,635 @@
+        num_tasks_sample (int): Number of random tasks to obtain data for each
+            iteration.
+        num_steps_prior (int): Number of transitions to obtain per task with
+            z ~ prior.
+        num_steps_posterior (int): Number of transitions to obtain per task
+            with z ~ posterior.
+        num_extra_rl_steps_posterior (int): Number of additional transitions
+            to obtain per task with z ~ posterior that are only used to train
+            the policy and NOT the encoder.
+        num_steps_per_eval (int): Number of transitions to evaluate on.
+        batch_size (int): Number of transitions in RL batch.
+        embedding_batch_size (int): Number of transitions in context batch.
+        embedding_mini_batch_size (int): Number of transitions in mini context
+            batch; should be same as embedding_batch_size for non-recurrent
+            encoder.
+        max_path_length (int): Maximum path length.
+        discount (float): RL discount factor.
+        replay_buffer_size (int): Maximum samples in replay buffer.
+        reward_scale (int): Reward scale.
+        update_post_train (int): How often to resample context when obtaining
+            data during training (in trajectories).
+        eval_deterministic (bool): Whether to make policy deterministic during
+            evaluation.
+
+    """
+
+    # pylint: disable=too-many-statements
+    def __init__(self,
+                 env,
+                 policy_class,
+                 inner_policy,
+                 encoder_class,
+                 qf,
+                 vf,
+                 num_train_tasks,
+                 num_test_tasks,
+                 latent_dim,
+                 encoder_hidden_sizes,
+                 policy_lr=3E-4,
+                 qf_lr=3E-4,
+                 vf_lr=3E-4,
+                 context_lr=3E-4,
+                 policy_mean_reg_coeff=1E-3,
+                 policy_std_reg_coeff=1E-3,
+                 policy_pre_activation_coeff=0.,
+                 soft_target_tau=0.005,
+                 kl_lambda=.1,
+                 optimizer_class=torch.optim.Adam,
+                 recurrent=False,
+                 use_information_bottleneck=True,
+                 use_next_obs_in_context=False,
+                 meta_batch_size=64,
+                 num_steps_per_epoch=1000,
+                 num_initial_steps=100,
+                 num_tasks_sample=100,
+                 num_steps_prior=100,
+                 num_steps_posterior=0,
+                 num_extra_rl_steps_posterior=100,
+                 num_steps_per_eval=1000,
+                 batch_size=1024,
+                 embedding_batch_size=1024,
+                 embedding_mini_batch_size=1024,
+                 max_path_length=1000,
+                 discount=0.99,
+                 replay_buffer_size=1000000,
+                 reward_scale=1,
+                 update_post_train=1,
+                 eval_deterministic=True):
+
+        self.env = env
```

Comment:
is there a reason this can't be private?

Reply:
I will make this private.